src/misc/binaryheap.hpp

```diff
@@ -50,27 +52,18 @@
 template <class T>
 class CBinaryHeapT {
 private:
-	uint items;    ///< Number of items in the heap
-	uint capacity; ///< Maximum number of items the heap can hold
-	T **data;      ///< The pointer to the heap item pointers
+	size_t items = 0; ///< Number of valid items in the heap
+	std::vector<T *> data; ///< The pointer to the heap item pointers
 
 public:
 	/**
 	 * Create a binary heap.
-	 * @param max_items The limit of the heap
+	 * @param initial_size The initial size of the heap
```

Comment:
Not its actual size, just reserved.

Reply:
Renamed it to `initial_capacity`. After all, it was not a limit because it would just reallocate if more space was needed.